storage/retry_conformance_test.go

```diff
@@ -683,12 +698,25 @@ func (et *emulatorTest) create(instructions map[string][]string) {
 	}
 
 	et.id = testRes.TestID
-
-	// Create wrapped client which will send emulator instructions
 	et.host.Path = ""
-	client, err := wrappedClient(et.T, et.id)
-	if err != nil {
-		et.Fatalf("creating wrapped client: %v", err)
+
+	// Create wrapped client which will send a retry test id via a constant header
+	var client *Client
+	if transport == "json" {
+		client, err = wrappedClient(et.T, et.id)
```

Comment:
I think we can actually rip out this whole custom client logic now that we added support for sending custom headers via the context: https://pkg.go.dev/cloud.google.com/go/storage#hdr-Sending_Custom_Headers . Works for both gRPC and JSON. The context should be applied per method call, not per client.

Reply:
Awesome, thanks! I removed the wrappedClient and RoundTrip per your comment. With the latest custom headers support, we're now attaching the retry test id per test method call.